awslabs/multi-model-server
# performance regression suite

Contributors: @dhruvaray @prashantsail, @maheshambule

## Issue #, if available:

## Description of changes:
It adds a performance monitoring test suite. This suite enables to quickly add test cases and specify pass fail criteria for different metrics. It creates a standard XML and HTML report with JUnit standard specifications. Refer README for more details.

Components added:

- Test suite runner script
- Sample test cases
- Server monitoring script
- Local monitoring script
- README


TODO- 
1. Add more metrics including GPU metrics
2. Auto Threshold calculation, per environment threshold
3. Environment profiles
4. Add more tests
5. Enable test case variables such as host, port to be specified from suite runner script

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
